private/model/api/shape_marshal.go

```diff
@@ -200,8 +207,16 @@ func isShapeFieldsNested(loc string, s *Shape) bool {
 	return loc == "Body" && len(s.LocationName) != 0 && s.API.Metadata.Protocol == "rest-xml"
 }
 
+func QuotedFormatTime(s marshalShapeRef) string  {
+	if  s.Ref.API.Metadata.Protocol == "json" || s.Ref.API.Metadata.Protocol == "rest-json" && s.Location() == "Body" {
```

Comment:
could you add parentheses around the or clause its initially confusing reading the if condition with the OR/AND pairing.